src/aggregate/aggregate_exec.c

```diff
@@ -398,8 +398,10 @@ void sendChunk(AREQ *req, RedisModule_Reply *reply, size_t limit) {
 
 done_3:
     SearchResult_Destroy(&r);
-    if (rc != RS_RESULT_OK) {
-      req->stateflags |= QEXEC_S_ITERDONE;
+
+    if ((rc == RS_RESULT_TIMEDOUT && req->reqConfig.timeoutPolicy == TimeoutPolicy_Fail)
```

Comment:
Consider using != RS_RESULT_OK and add the && !(rc == RS_RESULT_TIMEDOUT && req->reqConfig.timeoutPolicy == TimeoutPolicy_Return)